### ISSUE TITLE
Add DOI links to Solr document

### ISSUE THREAD

**Issue by pbinkley** [ualbertalib/HydraNorth#1097]
From https://github.com/ualbertalib/HydraNorth/issues/1091#issuecomment-219857973 : search results page rendering is slowed by the need to discover DOI urls for each item. Since they aren't in the Solr index, they have to be retrieved from Fedora. So: add them to metadata.rb, and confirm that this fixes the performance hit.

**Comment by mbarnett** [comment]
If we wanted to get more leverage out of this change, we could drop download links off of the search result page for the time being. It would make a huge, immediate difference until we reindex everything so that DOI urls are consistently in Solr. 

Download links would still be available on the Show page for an item (where they aren't as costly because the GenericFile is already loaded)

**Comment by leahvanderjagt** [comment]
This might pose a usability problem for users, but if the gains are huge
re: search then I would consider approving that change.

Question: Are any of the changes we are discussing through this thread
going to have an effect on the length of time it takes ERA to display the
item view from an item link? That is also slow, so if it's harder for the
user to get to the download link due to the change Matt proposes then I do
worry about solving one usability problem but unfortunately also creating a
new one.


On Tue, May 17, 2016 at 3:46 PM, Matthew Barnett <notifications@github.com>
wrote:

> If we wanted to get more leverage out of this change, we could drop
> download links off of the search result page for the time being. It would
> make a huge, immediate difference until we reindex everything so that DOI
> urls are consistently in Solr.
>
> Download links would still be available on the Show page for an item
> (where they aren't as costly because the GenericFile is already loaded)
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/ualbertalib/HydraNorth/issues/1097#issuecomment-219864258>
>



-- 
___________________________________________________________




*Leah VanderjagtDigital Repository Services Librarian University of
Albertat. 780.492.3851 *
*leahv@ualberta.ca <leahv@ualberta.ca> *

**Comment by mbarnett** [comment]
I'm separately looking into other things that may speed up individual item pages.

I think dropping the download links from search results is worth it temporarily. The search page is 80% faster for me locally if we stop rendering them.

If we leave them in place but add doi links to the Solr metadata, there will be no change to /catalog speed until we do a full reindex of the repository.

**Comment by leahvanderjagt** [comment]
That is a really big difference. I think that if it is a temporary measure
then we should remove the download links from the search results. The users
will tell me (I hope) if they get confused or inconvenienced.

Leah

On Tue, May 17, 2016 at 4:37 PM, Matthew Barnett <notifications@github.com>
wrote:

> I'm separately looking into other things that may speed up individual item
> pages.
>
> I think dropping the download links from search results is worth it
> temporarily. The search page is 80% faster for me locally if we stop
> rendering them.
>
> If we leave them in place but add doi links to the Solr metadata, there
> will be no change to /catalog speed until we do a full reindex of the
> repository.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/ualbertalib/HydraNorth/issues/1097#issuecomment-219875173>
>



-- 
___________________________________________________________




*Leah VanderjagtDigital Repository Services Librarian University of
Albertat. 780.492.3851 *
*leahv@ualberta.ca <leahv@ualberta.ca> *

**Comment by mbarnett** [comment]
That makes sense. I've added a new issue to track reintroducing the links once the DOI link reindex is complete: https://github.com/ualbertalib/HydraNorth/issues/1098

**Comment by pbinkley** [comment]
From a quick parsing of the current apache logs, the sources of clicks on download links look like this:

```
   1541 catalog
   1177 collections
     53 dashboard
    109 downloads
   1220 files
 ```

So about 2/3 will be affected (catalog = search results, collections uses same display). But if we can deploy this just before the ARK run (which will force a full reindexing), it will be a brief change.

**Comment by pbinkley** [comment]
Ark reindex job is part of #977
